lightning/src/ln/outbound_payment.rs

```diff
@@ -468,34 +469,38 @@ impl OutboundPayments {
 		NS::Target: NodeSigner,
 		L::Target: Logger,
 		IH: Fn() -> InFlightHtlcs,
-		SP: Fn(&Vec<RouteHop>, &PaymentHash, &Option<PaymentSecret>, u64, u32, PaymentId,
+		SP: Fn(&Vec<RouteHop>, &PaymentHash, RecipientOnionFields, u64, u32, PaymentId,
 			&Option<PaymentPreimage>, [u8; 32]) -> Result<(), APIError>,
 	{
 		let preimage = payment_preimage
 			.unwrap_or_else(|| PaymentPreimage(entropy_source.get_secure_random_bytes()));
 		let payment_hash = PaymentHash(Sha256::hash(&preimage.0).into_inner());
-		self.send_payment_internal(payment_id, payment_hash, &None, Some(preimage), retry_strategy,
-			route_params, router, first_hops, inflight_htlcs, entropy_source, node_signer,
-			best_block_height, logger, pending_events, send_payment_along_path)
+		self.send_payment_internal(payment_id, payment_hash, recipient_onion, Some(preimage),
```

Comment:
My thinking was that this could potentially be misused, i.e. a user just defaults to using `secret_only` everywhere without reading the docs, but dunno how much we care about that

Reply:
`invoice_secret_only` might be clearer, but I'm fine with the deterrent of not having a payment secret